code.py

```diff
@@ -1,43 +1,122 @@
 # Beacon Software
 # Made in collaboration with Mr. Douglas Chin
 
-from time import sleep
 import board
 import adafruit_irremote
 import pulseio
 import digitalio
 import time
 import servercom
-from beacon_client import BeaconClient
-from microcontroller import watchdog as w
-from watchdog import WatchDogMode
-import indication
-import supervisor
+import json
+import enum
 
-try:
-    if supervisor.runtime.run_reason == supervisor.RunReason.STARTUP:
-        indication.blink_color(indication.GREEN)
-    elif supervisor.runtime.run_reason == supervisor.RunReason.SUPERVISOR_RELOAD:
-        indication.blink_color(indication.RED)
+CHECK_INTERVAL = 15
+    
+class MessageStatus(enum.Enum):
+    SCHEDULED    = "Scheduled"
+    TRANSMITTED  = "Transmitted"
+    TRANSMITTING = "Transmitting..."
+    FAILED       = "Failed"
+
+
+class Destination(enum.Enum):
+    INFRARED = "Infrared"
+    VISIBLE = "Visible"
+
+
+class Connection(servercom.Connection):
+    def get_next_message(self):
+        resp = self.request('GET', '/beacon/message/next_queued',
+            headers=['User-Agent: CircuitPythonDude']
+        )
+        # {
+        #     "id": <hex string representing the object id>,
+        #     "timestamp": <epoch timestamp as a decimal>,
+        #     "offset": <seconds from request until instant>,
+        #     "destination": <"Infrared"|"Visible">,
+        #     "intensity": <8-bit integer>,
+        #     "message": <the message as a UTF-8 str>
+        # }
+        return json.loads(resp[1])
+        
+
+    def update_message_status(self, object_id:str, status:MessageStatus):
+        return self.request(
+            'PUT',
+            f'/beacon/message/{object_id}',
+            json.dumps({"status": status}),
+            content_type = 'application/json',
+            headers=['User-Agent: CircuitPythonDude']
+        ).code == 201
+
+
+@contextmanager
+def connection():
+	c = Connection()
+	try:
+		yield c
+	finally:
+		c.close()
+
```

Comment:
CircuitPython does not implement this.
However, `servercom.Connection` already implements `__exit__`, so perhaps a context manager isn't necessary?
```suggestion
connection = Connection
```